src/psyclone/psyir/nodes/node.py

```diff
@@ -123,12 +259,16 @@ class Node(object):
     START_POSITION = 0
     # The list of valid annotations for this Node. Populated by sub-class.
     valid_annotations = tuple()
+    # Class attributes of this node textual representations
+    _children_valid_format = "*[Node]"
```

Comment:
Would it make sense to set these to None here and then you could check that they are set by subclasses - a sort of abstract attribute? Or are you happy with the default being Node and people can override if they want to (plus if they read the developers guide which will explain :-)).

Reply:
I like the `None` idea (there may be better solution for Python > 2 e.g. https://stackoverflow.com/a/53417582) but since we still support Python 2 I changed them to None for now.

Reply:
I am happy with None but we already use abc in PSyclone so I don't see why we could not use it if we wanted to.

Reply:
Because what we need here is "ABC.abstractclassmethod" which was added to python 3, and is not available in python 2.